src/main/java/org/osc/controller/nsc/restserver/api/NetworkElementApis.java

```diff
@@ -0,0 +1,133 @@
+/*******************************************************************************
+ * Copyright (c) Intel Corporation
+ * Copyright (c) 2017
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.osc.controller.nsc.restserver.api;
+
+import java.util.List;
+
+import javax.persistence.EntityManager;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.osc.controller.nsc.api.SampleSdnRedirectionApi;
+import org.osc.controller.nsc.entities.NetworkElementEntity;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.jdbc.DataSourceFactory;
+import org.osgi.service.jpa.EntityManagerFactoryBuilder;
+import org.osgi.service.transaction.control.TransactionControl;
+import org.osgi.service.transaction.control.jpa.JPAEntityManagerProviderFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Component(service = NetworkElementApis.class)
+@Path("/controller/{id}/networkElements")
+@Consumes({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
+@Produces({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
+public class NetworkElementApis {
+
+    private static Logger LOG = LoggerFactory.getLogger(InspectionPortApis.class);
+
+    private EntityManager em;
+
+    @Reference(target = "(osgi.local.enabled=true)")
+    private TransactionControl txControl;
+
+    @Reference(target = "(osgi.unit.name=nsc-mgr)")
+    private EntityManagerFactoryBuilder builder;
+
+    @Reference(target = "(osgi.jdbc.driver.class=org.h2.Driver)")
+    private DataSourceFactory jdbcFactory;
+
+    @Reference(target = "(osgi.local.enabled=true)")
+    private JPAEntityManagerProviderFactory resourceFactory;
+
+    @Path("/{elementId}")
```

Comment:
no need to add the id on create APIs. Same thing for create inspection hook and inspection ports. I dont think we use the entity id on the creation apis of the security manager sample plugin.

Reply:
@emanoelxavier , Id is used for creation of NetworkElementEntity as we discussed during one of the remarks earlier. For InspectionPort and InspectionHook, Id is not passed as it is auto generated and similar to as in security sample code. 
This is how it was designed for the networkelemententity.
@dmitryintel , please feedback why Id is not auto generated for networkelemententity? did you encounter any issue when auto generation is enabled?

Reply:
got it